provider/datagen/src/source.rs

```diff
@@ -3,15 +3,27 @@
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
 use crate::error::DatagenError;
+use icu_codepointtrie::TrieType;
 use icu_provider::DataError;
 use std::path::{Path, PathBuf};
 
 /// Bag of options for datagen source data.
-#[derive(Clone, Debug, Default)]
+#[derive(Clone, Debug)]
 #[non_exhaustive]
 pub struct SourceData {
     cldr_paths: Option<CldrPaths>,
     uprops_root: Option<PathBuf>,
+    trie_type: TrieType,
+}
+
+impl Default for SourceData {
+    fn default() -> Self {
+        Self {
+            cldr_paths: None,
+            uprops_root: None,
+            trie_type: TrieType::Small,
```

Comment:
Suggestion: if you implement `Default` on `TrieType` you can keep the derive.

Reply:
I thought about that. I think it's better if each individual call site specifies what it wants the default TrieType to be.

There is not a universally correct answer to "what is the default TrieType", but Datagen can say, "we default to small data by default".